include/cuco/detail/static_map.inl

```diff
@@ -358,6 +427,102 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
   }
 }
 
+template <typename Key, typename Value, cuda::thread_scope Scope, typename Allocator>
+template <typename Hash, typename KeyEqual>
+__device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::erase(
+  key_type const& k, Hash hash, KeyEqual key_equal) noexcept
+{
+  auto current_slot{initial_slot(k, hash)};
+
+  value_type const insert_pair =
+    make_pair<Key, Value>(this->get_erased_key_sentinel(), this->get_empty_value_sentinel());
+
+  while (true) {
+    auto existing_key   = current_slot->first.load(cuda::std::memory_order_relaxed);
+    auto existing_value = current_slot->second.load(cuda::std::memory_order_relaxed);
+
+    // Key doesn't exist, return false
+    if (detail::bitwise_compare(existing_key, this->get_empty_key_sentinel())) { return false; }
+
+    // Key exists, return true if successfully deleted
+    if (key_equal(existing_key, k)) {
+      if constexpr (cuco::detail::is_packable<value_type>()) {
+        auto slot = reinterpret_cast<
+          cuda::atomic<typename cuco::detail::pair_converter<value_type>::packed_type>*>(
+          current_slot);
+        cuco::detail::pair_converter<value_type> expected_pair{
+          cuco::make_pair(existing_key, existing_value)};
+        cuco::detail::pair_converter<value_type> new_pair{insert_pair};
+
+        return slot->compare_exchange_strong(
+          expected_pair.packed, new_pair.packed, cuda::std::memory_order_relaxed);
```

Comment:
There should be a way to clean this up. Most semantics are covered by `packed_cas` already.

Reply:
Yeah, it is a bit messy. The only issue is that the current CAS functions like `packed_cas` are tailored for `insert` (the CAS is looking for an `empty_value_sentinel` and they return an `insert_result`.

Reply:
For `insert`, we insert an input key/value pair to replace empty key and empty value sentinels while for `erase`, we insert erase key and empty value sentinels to replace the slot content. There should be common things shared between `insert` and `erase`.